fw/http.c

```diff
@@ -5410,6 +5421,11 @@ tfw_h2_resp_adjust_fwd(TfwHttpResp *resp)
 
 	tfw_h2_msg_transform_setup(mit, resp->msg.skb_head, true);
 
+	r = tfw_hpack_enc_tbl_size_add(resp, last->data + last->len,
+				       &bytes_written);
+	if (unlikely(r))
+		goto clean;
```

Comment:
In case of `r != 0` we can not add SETTINGS ACK for the current frame & response and just send the response w/o it, which is prohibited by RFC 9113 6.5.3.

The other question about the approach to try to send SETTINGS ACK until it's sent is what happens if we can't send a response (we discussed this on the previous call)? We have to either send another (error) response or terminate the connection or stream. We send only one error response, so probably this approach can be simpler and more robust and efficient.

The current approach is fragile in the way that we have to call `tfw_hpack_enc_tbl_size_release()` and if we don't do this in any work flow, then we'll be calling the skb fragmentation for a while (this is an effort for #1103). In general, it's not good to introduce many inter-layer calls. We already have [very  sophisticated and fragile code](https://github.com/tempesta-tech/tempesta/issues/687#issuecomment-462445599) in http.c.

It seems the framing layer (e.g. `tfw_h2_make_frames()`) is called the last before the network transmission - can't we add the SETTINGS ACK there using a state in `TfwH2Ctx`?